### ISSUE TITLE
Long-lived TCP sessions for IRC + unstable connection after your default route = inconsistent 

### ISSUE THREAD

**Issue by Xe** [hexchat/hexchat#1534]
The big key issue here is that the underlying socket can go away without hexchat realizing until the socket is written to.

Reproduction will be easy provided you have access to public transit like Caltrain or BART and some way to get cellular internet.

**Comment by Arnavion** [comment]
That's what net_ping_timeout is for

**Comment by chadaustin** [comment]
I don't think net_ping_timeout is related.  Or, at least, it shouldn't be.  When I put my laptop to sleep at night and then wake it up in the morning (on a different wifi network), HexChat doesn't try to reconnect as it should.  Instead it just sits there indefinitely.  (I haven't waited minutes to see if it eventually disconnects.)

Fedora 22, Dell XPS 9343

**Comment by Arnavion** [comment]
Yes, and when it sends a ping it should either be told by the OS that the connection is broken, or receive no response from the server after the ping timeout. In either case it will reconnect.

You don't have to wait "minutes" for it if you set net_ping_timeout to a smaller value.
